purchase.py

```diff
@@ -23,3 +23,10 @@ def quote(cls, purchases):
                         'product_purchase_homologation.cannot_end_purchase',
                         purchase=purchase.id,
                         line=line.rec_name))
+                if (line.product.purchase_homologation_required and
```

Comment:
You must check if line.product is set before checking it's purchase_homologation_required field. Otherwise it will crash if no product is set.

Reply:
Also remember to check `line.type == 'line'` or do a continue in case `line.type != 'line'`